File: Day13/ArrProductSign.py
class Solution(object):
    def arraySign(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        sign = 1
        for i in range(len(nums)):
            if nums[i] == 0:
                sign = 0
            elif nums[i] < 0:
                sign *= -1
            else:
                sign *= 1
        return sign

if __name__ == "__main__":
    nums = list(map(int, input().split()))
    print(Solution().arraySign(nums))

File: Day13/ArrProductSign.java
import java.util.*;

class ArrProductSign {
    public int arraySign(int[] nums) {
        int sign = 1;
        for (int num : nums) {
            if (num == 0)
                return 0;
            else if (num < 0)
                sign *= -1;
        }
        return sign;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String[] parts = sc.nextLine().trim().split("\\s+");
        int[] nums = new int[parts.length];
        for (int i = 0; i < parts.length; i++) {
            nums[i] = Integer.parseInt(parts[i]);
        }
        System.out.println(new ArrProductSign().arraySign(nums));
    }
}
